include/rmm/cuda_stream_pool.hpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <rmm/cuda_stream.hpp>
+#include <rmm/cuda_stream_view.hpp>
+
+#include <atomic>
+#include <vector>
+
+namespace rmm {
+
+/**
+ * @brief A pool of CUDA streams.
+ *
+ * Provides fast access to a circular buffer of `cuda_stream` objects.
```

Comment:
I think this is an implementation detail we don't need to advertise as this point. It very well could change.

Reply:
People may want to know that it is fast. People might also want to know that they might get the same stream and therefore have false dependencies?

Reply:
How about:

```
Provides efficient access to collection of CUDA stream objects. Successive calls may return a handle to identical streams. For example, a possible implementation would be to maintain a circular buffer of stream objects. 
```

I just don't want to be locked into this always being a circular buffer.

Reply:
Done.